sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -712,6 +716,15 @@ def fit(self, X, y=None):
         self : object
             Returns self.
         """
+        if self.copy == "warn":
+            warn(
+                "The default value of `copy` will change from False to True in 1.10",
+                FutureWarning,
+            )
+            self._copy = False
```

Comment:
since this is only used in `fit` here, no need to set an attribute, could simply be a local variable.